src/internal/clusterstate/v2.8.0/pfsdb.go

```diff
@@ -249,6 +247,17 @@ func alterCommitsTable(ctx context.Context, tx *pachsql.Tx) error {
 	return nil
 }
 
+func alterCommitsTablePostDataMigration(ctx context.Context, env migrations.Env) error {
+	query := `
+	ALTER TABLE pfs.commits
+	    DROP CONSTRAINT fk_col_commit;
```

Comment:
Why drop this fk constraint?

Reply:
Because it forces us to make sure we need to have a corresponding collections.commits entry for each pfs.commits object.

Reply:
Setting the commit_id as nil makes the api not correct. What is the reason for keeping the constraint? We need to drop it once the migration is complete anyways.

Reply:
So I thought about this, and you are right to not set `commit_id` as null. I think what you are doing right now, i.e. `CommitID: CommitKey(commitInfo.Commit)` is correct.

I still think it's too early to drop this constraint. What you can do is update the PutHook for the commits collection to use your new `CreateCommit` function instead of the old `AddCommit`. See  https://github.com/pachyderm/pachyderm/blob/master/src/internal/pfsdb/pfsdb.go#L104-L110